hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java

```diff
@@ -158,10 +162,14 @@ private Comparator<ContainerID> orderContainersByUsedBytes() {
    * Checks whether a Container has the ReplicationType
    * {@link HddsProtos.ReplicationType#EC}.
    * @param container container to check
-   * @return true if the ReplicationType is EC, else false
+   * @return true if the ReplicationType is EC and "hdds.scm.replication
+   * .enable.legacy" is true, else false
    */
   private boolean isECContainer(ContainerInfo container) {
-    return container.getReplicationType().equals(HddsProtos.ReplicationType.EC);
+    boolean legacyEnabled = ozoneConfiguration.
+        getBoolean("hdds.scm.replication.enable.legacy", true);
```

Comment:
The same config can be looked up by `replicationManager.getConfig().isLegacyEnabled()`.  That way we could avoid the need to pass `OzoneConfiguration` to the constructor, as well as the duplication of config key name.

Nit: I think we can also skip storing this in a local variable.

Reply:
Okay, I'm using `replicationManager.getConfig().isLegacyEnabled()` in this class and `ContainerBalancerTask` as well now. Changed the unit tests accordingly.